### ISSUE TITLE
Init cookie issue.

### ISSUE THREAD

**Issue by hcaihao** [bungle/lua-resty-session#27]
`local session = require "resty.session".start{ secret = "623q4hR325t36VsCD3g567922IC0073T" }`

```
local session = require "resty.session".new()
session.secret = "623q4hR325t36VsCD3g567922IC0073T"
```

I use thie first line to init session is correct, but use the second two lines get an error:
`session.lua:86: attempt to concatenate local 'i' (a nil value)`

Is this a bug?

**Comment by hcaihao** [comment]
BTW, How to init cookie with some options with start, like:

local session = require "resty.session".start{ secret = "623q4hR325t36VsCD3g567922IC0073T", persistent = true, lifetime = 86400 }

**Comment by bungle** [comment]
For the second question:

```lua
local session = require "resty.session".start{ secret = "623q4hR325t36VsCD3g567922IC0073T", cookie = { persistent = true, lifetime = 86400 } }
```

**Comment by bungle** [comment]
I will check that error you are having, it seems to be a bug or something. Btw, I'm also implementing SameSite-cookie flag.

**Comment by bungle** [comment]
Can you give me a full example where this error occurs. It seems you are also calling save?

**Comment by hcaihao** [comment]
OK, It's my mistake, I forget to call start, the code runs well now! Tks.

**Comment by bungle** [comment]
I just released a new version 2.4 where this should be fixed (well, I still consider it a bug in a code, but now it at least doesn't crash anymore, if you don't start, but remember that it will overwrite the session if it was started on previous request).

**Comment by bungle** [comment]
I think this is fixed now on the latest 2.5. I will close this now. Please reopen if you still feel there is a problem in this. Thanks!
